### DISCUSSION TITLE
response buttons to appear only after video finishes [jspsych-video-button-response plugin]

### DISCUSSION THREAD

**Discussion by mgorenstein** [jspsych/jsPsych#1164] [Q&A] [answered]
Hi there - I'm wondering whether there's a recommended way to delay the visibility of the response buttons until the conclusion of the video stimulus in the `jspsych-video-button-response` plugin. That is, the participant wouldn't be able to see the buttons or make a response until the video has finished playing. 

Otherwise, I see that there's a `trial_ends_after_video` flag which functions somewhat similarly, so maybe I could modify the plugin to follow that logic and, say, change the visibility of the button elements following a video end event.

Thanks!

Mark

**Comment by becky-gilbert** [comment, marked as answer]
Hi Mark, you're in luck - we just decided to add the new `response_allowed_while_playing` parameter to the video and audio plugins, but it hasn't been merged into the library yet (#1145). For now, you can find the video-button-response plugin with this parameter here: https://github.com/becky-gilbert/jsPsych/blob/respond-while-playing/plugins/jspsych-video-button-response.js This parameter is `true` by default, but you can change it to `false` to disable the response buttons until the video ends. Or, if you want to hide the buttons instead of disable them, you can modify the code ([line 203](https://github.com/becky-gilbert/jsPsych/blob/8b5a68edfc93784907b6d08d373d42f11292fef0/plugins/jspsych-video-button-response.js#L203) and [line 230](https://github.com/becky-gilbert/jsPsych/blob/8b5a68edfc93784907b6d08d373d42f11292fef0/plugins/jspsych-video-button-response.js#L230)) so that the button's `style.visibility` property is set to `visible`/`hidden`, instead of setting its `disabled` property to `true`/`false`.

**Reply by mgorenstein** [reply]
Oh, many thanks, Becky, this is exactly what I had in mind.
